1 - Conversão de graus Celsius para Fahrenheit – Crie um programa que converta graus
Celsius em Fahrenheit. A fórmula é a seguinte:

F = 9/5*C + 32


O programa deve solicitar ao usuário que insira uma temperatura em graus Celsius e, em
seguida, exiba a temperatura convertida em Fahrenheit. Após construir esse programa,
modifique-o para que converta graus Fahrenheit em graus Celsius.


In [4]:
unidade_medida = input("Digite a unidade de medida atual (C - Celcius/F - Fahrenheit): ").upper()
temperatura = float(input("Digite o valor a ser convertido: "))
if unidade_medida == "C":
    valor_convertido = (9 * temperatura) / 5 + 32
    unidade_convertida = "F"
else:
    valor_convertido = (temperatura - 32) * 5 / 9
    unidade_convertida = "C"

print(f"{temperatura:.2f}°{unidade_medida} equivale a {valor_convertido:.2f}°{unidade_convertida}.")


15.00°C equivale a 59.00°F.


2- Escreva um programa que receba um número e escreva “Par” caso esse número seja par
e escreva “Impar” caso esse número seja impar.


In [12]:
num = int(input('Digite um número: '))

if not num&1:
    print(f'{num} é par')
else:
    print(f'{num} é ímpar')

4 é par


3 - Escreva um programa que receba dois números, exiba as opções:

1. adição
2. subtração

Então peça ao usuário para escolher uma das opções. Caso escolha a opção 1 o programa
deve realizar a soma dos dois números lidos e exibir. Caso escolha a opção 2 o programa
deve realizar a subtração dos dois números lidos e exibir.

In [13]:
numero1 = float(input("Digite o primeiro número: "))
numero2 = float(input("Digite o segundo número: "))

operacao = input("Digite a operação: 1 - Adição | 2 - Subtração): ")

if operacao == "1":
    resultado = numero1 + numero2 
elif operacao == "2":
    resultado = numero1 - numero2
else:
    print('Operação inválida!')

print(f"{numero1} {operacao} {numero2} = {resultado:.2f}")

5.0 1 6.0 = 11.00


4 - Numa determinada escola, os critérios de aprovação são os seguintes:
- O aluno deve ter, no máximo, 25% de faltas;
- A nota final deve ser igual ou superior a 7,00.
Construa um programa para ler as notas que um aluno tirou nos 4 bimestres, o número total
de aulas e o número de faltas, mostrando ao final a situação do aluno como sendo
“Aprovado”, “Reprovado por Faltas” e “Reprovado por média”, considerando que a
reprovação por faltas se sobrepõe a reprovação por nota.
5 - Após construir o programa anterior, crie mais duas versões dele para prever as seguintes
situações:
- Um aluno pode ficar em recuperação se possuir frequência suficiente (superior a
75%) e média superior a 5 mas inferior a 7;
- Caso um aluno reprove por média e faltas, sua situação deve ser “Reprovado por
Média e Faltas” (ao invés de simplesmente “Reprovado por Faltas” como antes).

In [ ]:
nota_1 = float(input("Digite a nota do 1º bimestre: "))
nota_2 = float(input("Digite a nota do 2º bimestre: "))
nota_3 = float(input("Digite a nota do 3º bimestre: "))
nota_4 = float(input("Digite a nota do 4º bimestre: "))

media_final = (nota_1 + nota_2 + nota_3 + nota_4) / 4

total_aulas = int(input("Digite o número total de aulas: "))
total_faltas = int(input("Digite o número de faltas: "))

frequencia = total_faltas / total_aulas <= 0.25

is_aprovado_por_media = media_final >= 7.0

if total_faltas / total_aulas > 0.25:
    situacao_aluno = "Reprovado por Faltas"
elif is_aprovado_por_media and frequencia:
    situacao_aluno = "Aprovado"
elif frequencia:
    situacao_aluno = "Recuperação"
else:
    situacao_aluno = "Reprovado por Média e Faltas"

print(f"Resultado do aluno: {situacao_aluno}")


5 - Após construir o programa anterior, crie mais duas versões dele para prever as seguintes
situações:
- Um aluno pode ficar em recuperação se possuir frequência suficiente (superior a
75%) e média superior a 5 mas inferior a 7;
- Caso um aluno reprove por média e faltas, sua situação deve ser “Reprovado por
Média e Faltas” (ao invés de simplesmente “Reprovado por Faltas” como antes).

In [17]:
nota_1 = float(input("Nota do 1º bimestre: "))
nota_2 = float(input("Nota do 2º bimestre: "))
nota_3 = float(input("Nota do 3º bimestre: "))
nota_4 = float(input("Nota do 4º bimestre: "))

# Cálculo da média final
media_final = (nota_1 + nota_2 + nota_3 + nota_4) / 4

# Coleta de dados de faltas e aulas
numero_aulas = int(input("Número total de aulas: "))
numero_faltas = int(input("Número de faltas: "))

# Verificação da frequência
frequencia_suficiente = numero_faltas / numero_aulas <= 0.25

# Verificação da aprovação por média
aprovado_por_media = media_final >= 7.0

# Determinação da situação final
if numero_faltas / numero_aulas > 0.25:
  situacao = "Reprovado por Faltas"
elif aprovado_por_media and frequencia_suficiente:
  situacao = "Aprovado"
elif frequencia_suficiente and media_final > 5:
  situacao = "Recuperação"
else:
  situacao = "Reprovado por Média e Faltas"

# Impressão da situação final
print(f"Situação: {situacao}")

Situação: Reprovado por Média e Faltas


6 - Escreva um programa que peça ao usuário para fornecer um dia, mês e o ano arbitrários
e determine se esses dados correspondem a uma data válida. Não deixe de considerar que
existem meses com 30 e 31 dias, e que fevereiro pode ter 28 ou 29 dias, dependendo se o
ano for bissexto. Considere que um ano é bissexto quando for divisível por 4.

In [ ]:
dia = int(input("Digite um dia: "))
mes = int(input("Digite um mês: "))
ano = int(input("Digite um ano: "))

if ano%4 == 0:
    if mes == 1:
        if dia 
    elif mes == 2:
    elif mes == 3:
    elif mes == 4:
    elif mes == 5:
    elif mes == 6:
    elif mes == 7:
    elif mes == 9:
    elif mes == 10:
    elif mes == 11:
    elif mes == 12:

7- Construa um programa que leia uma data qualquer (dia, mês e ano) e calcule a data do
próximo dia. Lembre-se que em anos bissextos o mês de fevereiro tem 29 dias. Lembre-se
que um ano é bissexto quando for divisível por 4.

8 - Faça um programa que leia as duas notas parciais obtidas por um aluno numa
disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos
obedece à tabela abaixo:
Média de Aproveitamento
- Entre 9.0 e 10.0 Conceito - A
- Entre 7.5 e 8.9 - B
- Entre 6.0 e 7.4 - C
- Entre 4.0 e 5.9 - D
- Entre 0 e 3.9 - E

O programa deve mostrar na tela as notas, a média, o conceito correspondente e a
mensagem:
APROVADO se o conceito for A, B ou C.
REPROVADO se o conceito for D ou E.

9 - As Organizações XTC resolveram dar um aumento de salário aos seus colaboradores e
lhe contrataram para desenvolver o programa que calcula os reajustes. Faça um programa
que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado
no salário atual:
- salários até R$ 280,00 (incluindo): aumento de 20%
- salários entre R$ 280,00 e R$ 700,00: aumento de 15%
- salários entre R$ 700,00 e R$ 1500,00: aumento de 10%
- salários de R$ 1500,00 em diante: aumento de 5%
Após o aumento ser realizado informe na tela:
- o salário antes do reajuste;
- o percentual de aumento aplicado;
- o valor do aumento;
- o novo salário, após o aumento.